assets/js/googlesitekit/datastore/user/utils.test.js

```diff
@@ -0,0 +1,202 @@
+			expect( getDateString( gmtString ) ).toEqual( '2020-01-01' );
+		} );
+		it( `should return "2020-01-01" for UTC string (${ utcString }) value of the same date`, () => {
+			expect( getDateString( utcString ) ).toEqual( '2020-01-01' );
+		} );
+	} );
+
+	describe( 'getPreviousDate', () => {
+		it( 'should throw error if no param is passed', () => {
+			try {
+				getPreviousDate();
+			} catch ( error ) {
+				expect( error.message ).toEqual( INVALID_DATE_INSTANCE_ERROR );
+			}
+		} );
+		it( 'should throw error if date supplied is invalid', () => {
+			try {
+				getPreviousDate( 'invalid-date', 1 );
+			} catch ( error ) {
+				expect( error.message ).toEqual( INVALID_DATE_INSTANCE_ERROR );
+			}
+		} );
+		it( 'should throw error if date supplied is invalid date', () => {
+			try {
+				getPreviousDate( '2020-99-99', 1 );
+			} catch ( error ) {
+				expect( error.message ).toEqual( INVALID_DATE_INSTANCE_ERROR );
+			}
+		} );
+		it( 'should return "2020-01-01" for a date of "2020-01-01" and a days before value of 0', () => {
+			expect( getPreviousDate( '2020-01-01', 0 ) ).toEqual( '2020-01-01' );
+		} );
+		it( 'should return "2020-01-01" for a date of "2020-01-02" and a days before value of 1', () => {
+			expect( getPreviousDate( '2020-01-02', 1 ) ).toEqual( '2020-01-01' );
+		} );
+		it( 'should return "2020-01-01" for a date of "2020-01-08" and a days before value of 7', () => {
+			expect( getPreviousDate( '2020-01-08', 7 ) ).toEqual( '2020-01-01' );
+		} );
+		it( 'should return "2020-01-01" for a date of "2020-02-01" and a days before value of 31', () => {
+			expect( getPreviousDate( '2020-02-01', 31 ) ).toEqual( '2020-01-01' );
+		} );
+		it( 'should return "2020-01-01" for a date of "2021-01-01" and a days before value of 366', () => {
+			expect( getPreviousDate( '2021-01-01', 366 ) ).toEqual( '2020-01-01' );
+		} );
+		it( 'should return "2019-12-31" for a date of "2020-01-01" and a days before value of 1', () => {
+			expect( getPreviousDate( '2020-01-01', 1 ) ).toEqual( '2019-12-31' );
+		} );
+	} );
+
+	describe( 'getPreviousWeekDate', () => {
+		it( 'should throw error if no param is passed', () => {
+			try {
+				getPreviousWeekDate();
+			} catch ( error ) {
+				expect( error.message ).toEqual( INVALID_DATE_INSTANCE_ERROR );
+			}
+		} );
+		it( 'should throw error if date supplied is invalid', () => {
+			try {
+				getPreviousWeekDate( 'invalid-date', 1 );
+			} catch ( error ) {
+				expect( error.message ).toEqual( INVALID_DATE_INSTANCE_ERROR );
+			}
+		} );
+		it( 'should throw error if date supplied is invalid date', () => {
+			try {
+				getPreviousWeekDate( '2020-99-99', 1 );
+			} catch ( error ) {
+				expect( error.message ).toEqual( INVALID_DATE_INSTANCE_ERROR );
+			}
+		} );
+		it( 'should go back 7 days with daysBefore value of 0 and date of "2020-01-30"', () => {
+			expect( getPreviousWeekDate( '2020-01-30', 0 ) ).toEqual( '2020-01-23' );
```

Comment:
See other comment, I'd expect this to return the exact same date if the "offset" is `0`.